source/StartConditionsPanel.cpp

```diff
@@ -0,0 +1,292 @@
+		font.Draw(name, point, isHighlighted ? bright : medium);
+		point += Point(0., entryBox.Height());
+	}
+}
+
+
+
+bool StartConditionsPanel::Drag(double dx, double dy)
+{
+	if(entryListBox.Contains(hoverPoint))
+	{
+		listScroll -= dy;
+		listScroll = min(entryBox.Height() * (GameData::StartOptions().size()-1), listScroll); 
+		listScroll = max(0., listScroll);
+	}
+	// TODO: When #4123 gets merged, re-add scrolling support.
+	// Right now it's pointless because it would make the text overflow.
+	
+	return true;
+}
+
+
+
+bool StartConditionsPanel::Scroll(double dx, double dy)
+{
+	return Drag(0., dy * Preferences::ScrollSpeed());
+}
+
+
+
+bool StartConditionsPanel::Hover(int x, int y)
+{
+	hoverPoint = Point(x, y);
+	return true;
+}
+
+
+
+bool StartConditionsPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, bool isNewPress)
+{
+	
+	if(key == 'b' || key == SDLK_ESCAPE || command.Has(Command::MENU) || (key == 'w' && (mod & (KMOD_CTRL | KMOD_GUI))))
+		GetUI()->Pop(this);
+	else if(key == SDLK_UP || key == SDLK_DOWN || key == SDLK_PAGEUP || key == SDLK_PAGEDOWN) 
+	{
+		int offset = 0;
+		offset += (key == SDLK_DOWN) - (key == SDLK_UP);
+		offset += PAGE_INTERVAL * ((key == SDLK_PAGEDOWN) - (key == SDLK_PAGEUP));
+		
+		listScroll += offset * entryBox.Height();
+		listScroll = min(max(listScroll, 0.), (GameData::StartOptions().size() - 1) * entryBox.Height());
+		
+		if(GameData::StartOptions().begin() - chosenStartIterator < offset)
+			chosenStartIterator = GameData::StartOptions().begin();
+		else if(GameData::StartOptions().end() - chosenStartIterator < offset)
+			chosenStartIterator = GameData::StartOptions().end() - 1;
+		else
+			chosenStartIterator += offset;
+		
+		chosenStart = &*chosenStartIterator;
+		descriptionWrappedText.Wrap(chosenStart->GetDescription());
+		
+		return false;
+	}
+	else if(key == 's' || key == 'n' || key == '\n')
+	{
+		if(!chosenStart)
+			return true;
+		
+		player.New(*chosenStart);
+		
+		if(chosenStart->GetConversation().IsEmpty())
+		{
+			// If no conversation was defined, then skip the conversation panel.
+			OnCallback(0);
+		}
+		else
+		{
+			ConversationPanel *panel = new ConversationPanel(
+				player, chosenStart->GetConversation());
+			GetUI()->Push(panel);
+			panel->SetCallback(this, &StartConditionsPanel::OnCallback);
+		}
+	}
+	else 
+		return true;
+	return false;
+}
+
+
+
+bool StartConditionsPanel::Click(int x, int y, int clicks)
+{
+	// Check it's inside of the entry list box.
+	if(!entryListBox.Contains(Point(x, y)))
+		return false;
+	
+	for(const auto &it : startConditionsClickZones)
+	{
+		if(it.Contains(Point(x, y + listScroll)))
+		{
+			chosenStartIterator = it.Value();
+			
+			if(chosenStart != &*chosenStartIterator)
+				descriptionScroll = 0;
+			chosenStart = &*chosenStartIterator;
+			descriptionWrappedText.Wrap(chosenStart->GetDescription());
+			break;
+		}
+	}
+	
+	if(chosenStart)
+		descriptionWrappedText.Wrap(chosenStart->GetDescription());
```

Comment:
We're wrapping the new description when the selected item changes, and then unconditionally wrapping it again here -- I expect only one of them is needed (probably the conditional one).